src/slider.tsx

```diff
@@ -1,164 +1,428 @@
+                            setIsFocused(false);
+                            thumbRef.current?.blur();
+                        }
+                        break;
+                    default:
+                    // do nothing
                 }
+            };
+
+            const handleKeyUp = () => (isTabKeyDownRef.current = false);
+
+            document.addEventListener('keydown', handleKeyDown, false);
+            document.addEventListener('keyup', handleKeyUp, false);
+            return () => {
+                document.removeEventListener('keydown', handleKeyDown, false);
+                document.removeEventListener('keyup', handleKeyUp, false);
+            };
+        }, [isFocused]);
+
+        const handleKeyDown = (e: React.KeyboardEvent<HTMLDivElement>) => {
+            switch (e.keyCode) {
+                case RIGHT:
+                case UP:
+                    cancelEvent(e);
+                    handleChange(finalValue + step, false);
+                    break;
+                case LEFT:
+                case DOWN:
+                    cancelEvent(e);
+                    handleChange(finalValue - step, false);
+                    break;
+                case HOME:
+                    cancelEvent(e);
+                    handleChange(min, false);
+                    break;
+                case END:
+                    cancelEvent(e);
+                    handleChange(max, false);
+                    break;
+
+                default:
+                // do nothing
             }
+        };
 
-            setValueRanger(ranger);
-            setStep(steps);
-        }
-    }, [steps, max, min, getApproximation, value, step, getOrder]);
+        const isTouchableDevice = isClientSide() ? window.matchMedia('(pointer: coarse)').matches : false;
+        const thumbSize = isIos ? IOS_THUMB_SIZE : DEFAULT_THUMB_SIZE;
+        const touchableArea = isTouchableDevice
+            ? MOBILE_TOUCHABLE_AREA
+            : isIos
+            ? IOS_TOUCHABLE_AREA
+            : DESKTOP_TOUCHABLE_AREA;
+
+        const updateCurrentValue = (pointerPosition: number) => {
+            const track = trackRef.current;
+            if (track) {
+                const leftBorder = track.getBoundingClientRect().left + thumbSize / 2;
+                const rightBorder = track.getBoundingClientRect().right - thumbSize / 2;
+                handleChange((pointerPosition - leftBorder) / (rightBorder - leftBorder), true);
+            }
+        };
 
-    return (
-        <section className={styles.container} aria-label={arialLabel}>
-            <div className={styles.targetContainer}>
+        const onPointerMove = (e: PointerEvent) => {
+            cancelEvent(e);
+            updateCurrentValue(e.clientX);
+        };
+
+        const capturePointerMove = (e: React.PointerEvent<HTMLDivElement>) => {
+            const thumb = thumbRef.current;
+            if (thumb) {
+                thumb.onpointermove = onPointerMove;
```

Comment:
This was intended, because if I write onPointerMove as a stable function, it will require handleChange() to be stable too, and we can't do that because handleChange depends on the function exposed by the form context (and that function is not memorized). If I do that, there will be an infinite loop of re-renders